legate/core/launcher.py

```diff
@@ -975,12 +983,15 @@ def build_single_task(self, argbuf: BufferBuilder) -> SingleTask:
             task.set_point(self._point)
         return task
 
-    def execute(self, launch_domain: Rect) -> FutureMap:
+    def execute(
+        self, launch_domain: Rect
+    ) -> tuple[FutureMap, dict[Store, LegionPartition]]:
         task = self.build_task(launch_domain, BufferBuilder())
         result = self._context.dispatch(task)
         assert isinstance(result, FutureMap)
         self._out_analyzer.update_storages()
-        return result
+        out_partitions = self._out_analyzer.collect_partitions()
+        return result, out_partitions
```

Comment:
dataclass could be nice here rather than a tuple

Reply:
https://github.com/nv-legate/legate.core/pull/699/commits/81bd87d21c59aaa6ad06f9db3b42dc5f8caf59d6